payments/paypal/__init__.py

```diff
@@ -310,15 +310,19 @@ def release(self, payment):
         self.post(payment, url)
 
     def refund(self, payment, amount=None):
-        if amount is None:
-            amount = payment.captured_amount
-        amount_data = self.get_amount_data(payment, amount)
-        refund_data = {"amount": amount_data}
+        refund_data = {}
+        if amount is not None:
+            refund_data["amount"] = self.get_amount_data(payment, amount)
         links = self._get_links(payment)
         url = links["refund"]["href"]
-        self.post(payment, url, data=refund_data)
+        response = self.post(payment, url, data=refund_data)
         payment.change_status(PaymentStatus.REFUNDED)
-        return amount
+        if response["amount"]["currency"] != payment.currency:
+            raise NotImplementedError(
+                f"refund's currency other than {payment.currency} not supported yet: "
+                f"{response['amount']['currency']}"
+            )
```

Comment:
This should be done before POST'ing, otherwise you refund but still raise.

Reply:
Can you be more specific, please? Before POST'ing, we do set the refund's currency to the payment's currency using `refund_data["amount"] = self.get_amount_data(payment, amount)`. This case only makes sure that PayPal does not return response in a different currency for some unexpected reason. Please, help me understand your suggestion.

Reply:
I misinterpreted this bit of line, disregard.